ffi/blitbuffer.lua

```diff
@@ -170,13 +172,13 @@ local P_BlitBuffer = ffi.typeof("BlitBuffer*")
 local P_BlitBuffer_ROData = ffi.typeof("const BlitBuffer*")
 
 -- metatables for color types:
-local Color4L_mt = {__index={}}
-local Color4U_mt = {__index={}}
-local Color8_mt = {__index={}}
-local Color8A_mt = {__index={}}
-local ColorRGB16_mt = {__index={}}
-local ColorRGB24_mt = {__index={}}
-local ColorRGB32_mt = {__index={}}
+local Color4L_mt = {__index={ is_rgb=false }}
+local Color4U_mt = {__index={ is_rgb=false }}
+local Color8_mt = {__index={ is_rgb=false }}
+local Color8A_mt = {__index={ is_rgb=false }}
+local ColorRGB16_mt = {__index={ is_rgb=false }}
+local ColorRGB24_mt = {__index={ is_rgb=false }}
+local ColorRGB32_mt = {__index={ is_rgb=true }}
```

Comment:
I'm... not necessarily a fan of this approach.

I get the idea of avoiding duplication like in @Galunid's PoC, but.... eeeh :/.

Possibly something based of `ffi.istype` instead?

Reply:
Initially I wanted to do something like `if color:type() == "ColorRGB32"` or similar but I wasn't sure how that interacts with ffi and I'm not super familiar with lua apis.
Would the expected syntax be `if ffi.istype(ColorRGB32, color)` ?

Reply:
Or `P_ColorRGB32` if it's a pointer, yeah.